mergify_engine/gh_pr_fullifier.py

```diff
@@ -103,78 +103,90 @@ def disabled_by_rules(pull, **extra):
+        github_desc = disabled
     elif reviews_ko:
-        weight = -1
-        status_desc = "Change requests need to be dismissed"
+        mergify_state = MergifyState.NOT_READY
+        github_state = "pending"
+        github_desc = "Change requests need to be dismissed"
     elif len(reviews_ok) < reviews_required:
-        weight = -1
-        status_desc = (
+        mergify_state = MergifyState.NOT_READY
+        github_state = "pending"
+        github_desc = (
             "%d/%d approvals required" %
             (len(reviews_ok), reviews_required)
         )
-    elif (pull.mergeable_state == "clean"
-          and pull.mergify_engine["combined_status"] == "success"):
-        # Best PR ever, up2date and CI OK
-        weight = 11
-        status_desc = "Will be merged soon"
     elif pull.mergeable_state in ["clean", "unstable"]:
-        weight = 10
-        status_desc = "Will be merged soon"
+        mergify_state = MergifyState.READY
+        github_state = "success"
+        github_desc = "Will be merged soon"
     elif (pull.mergeable_state == "blocked"
           and pull.mergify_engine["combined_status"] == "pending"):
-        # Maybe clean soon, or maybe this is the previous run
-        # selected PR that we just rebase
-        weight = 10
-        status_desc = "Waiting for CI success"
+        # Maybe clean soon, or maybe this is the previous run selected PR that
+        # we just rebase, or maybe not. But we set the mergify_state to
+        # ALMOST_READY to ensure we do not rebase multiple pull request in //
+        mergify_state = MergifyState.ALMOST_READY
+        github_state = "pending"
+        github_desc = "Waiting for CI success"
     elif pull.mergeable_state == "behind":
         # Not up2date, but ready to merge, is branch updatable
         if not pull.base_is_modifiable:
-            weight = -1
-            status_desc = ("Pull request can't be updated with latest "
+            mergify_state = MergifyState.NOT_READY
+            github_state = "failure"
+            github_desc = ("Pull request can't be updated with latest "
                            "base branch changes, owner doesn't allow "
                            "modification")
         elif pull.mergify_engine["combined_status"] == "success":
-            weight = 7
-            status_desc = ("Pull request will be updated with latest base "
+            mergify_state = MergifyState.NEED_BRANCH_UPDATE
+            github_state = "pending"
+            github_desc = ("Pull request will be updated with latest base "
                            "branch changes soon")
-        elif pull.mergify_engine["combined_status"] == "pending":
-            weight = 5
-            status_desc = "Waiting for CI success"
         else:
-            weight = -1
-            status_desc = "Waiting for CI success"
+            mergify_state = MergifyState.NOT_READY
+            github_state = "pending"
+            github_desc = "Waiting for CI success"
```

Comment:
ditto

use a variable  or class or something to DRY :)

Reply:
done